File: Sesion 11/Marco teorico/4LA EXCEPCIÓN ES CAPTURADA EN UN PUNTO ESPECÍFICO DEL PROGRAMA ( CATCH )/4.py
class OutOfRangeError(Exception):
    def __init__(self, message):
        self.message = message
        super().__init__(self.message)

def verificar(value):
    if value > 100:
        raise OutOfRangeError("Error: El valor es mayor a 100.")

try:
    num = int(input("Ingrese un número: "))
    verificar(num)
    print("El valor es:", num)
except OutOfRangeError as e:
    print("Excepción capturada:", e)
except ValueError:
    print("Error: Ingrese un número entero válido.")

File: Sesion 11/Marco teorico/4LA EXCEPCIÓN ES CAPTURADA EN UN PUNTO ESPECÍFICO DEL PROGRAMA ( CATCH )/4.cpp
#include <iostream>
#include <exception>
using namespace std;
void verificar(int value) {
    if (value > 100) {
        throw out_of_range("Error: El valor es mayor a 100.");
    }
}


int main() {
    int num;


    cout << "Ingrese un número: ";
    cin >> num;


    try {
        verificar(num);
        cout << "El valor es: " << num << endl;
    }
    catch (const out_of_range& e) {
        cerr << "Excepción capturada: " << e.what() << endl;
    }


    return 0;
}

